monadic-rx/shared/src/main/scala/mhtml/rx.scala

```diff
@@ -250,6 +264,22 @@ object Rx {
         Cancelable { () => cc.cancel; self.imitating = false }
       } else run(other)(effect)
 
+    case shareRx @ Sharing(self) =>
+      if (!shareRx.isSharing) {
+        shareRx.sharingCancelable = run(self){ x =>
+          shareRx.sharingMemo := x
+        }
+        shareRx.isSharing = true
+      }
+      val foreachCancelable = shareRx.sharingMemo.foreach(x => effect(x.asInstanceOf[A]))
+      Cancelable { () => {
+        foreachCancelable.cancel
+        if (shareRx.sharingMemo.subscribers.isEmpty) {
+          shareRx.sharingCancelable.cancel
```

Comment:
Don't forget to `shareRx.sharingCancelable = Cancelable.empty` here or it might leak something

BTW I wonder if you could get rid of `isSharing` and use `sharingCancelable eq Cancelable.empty` instead.

Reply:
I have to admit I've been lax about worrying about memory leaks on the JVM - maybe it has made me lazy, but I see your point, thanks! I made `isSharing` a method on `Sharing`. A note, for `eq` I got:

```
value eq is not a member of mhtml.Cancelable
[error] Note that Cancelable extends Any, not AnyRef.
[error] Such types can participate in value classes, but instances
[error] cannot appear in singleton types or in reference comparisons.
[error]     protected[Rx] def isSharing = !(sharingCancelable eq Cancelable.empty)
```

`==` seems to work.